src/network/local_socket_reply.cpp

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (C) 2020 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "local_socket_reply.h"
+
+#include <multipass/format.h>
+
+namespace mp = multipass;
+
+namespace
+{
+constexpr int len = 65536;
+
+// The following is mostly "borrowed" from Qt src/network/access/qhttpthreaddelegate.cpp
```

Comment:
Right, I just put the comment here to let anyone reading the code know where it came from.  Of course it's not necessary to put that, but I thought it would make it more clear.  Any additional opinions based on what I said or are you still on the notion to whack it?

Reply:
Then just give a URL to the file, saying something like "Status code mapping based on <url>".

Reply:
Ack, will do.

Reply:
Done